src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/executors/SensitiveTaskEventLogsExecutor.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.config.executors;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.PreDestroy;
+
+@Slf4j
+@Component
+@SuppressWarnings("PMD.DoNotUseThreads")
+public class SensitiveTaskEventLogsExecutor {
+
+    @Autowired
+    private ExecutorService sensitiveTaskEventLogsExecutorService;
+
+
+    @PreDestroy
+    public void cleanup() {
+        log.info("Shutting down SensitiveTaskEventLog executor");
+        sensitiveTaskEventLogsExecutorService.shutdown();
```

Comment:
I am guessing this is the same as with the Case Event Handler so should be ok.  But could we use awaitTermination in future?